File: frontend/src/api/dto/routeListResponse.ts
import type { RouteDto } from './routeDto.ts'

/**
 * Paginated route list response
 */
export interface RouteListResponse {
  /** List of routes */
  routes: RouteDto[]
  /** Total number of routes */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}

File: backend/src/main/java/fr/pedalons/dto/routes/response/RouteListResponse.java
package fr.pedalons.dto.routes.response;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated route list response")
@ValidateSchema
public record RouteListResponse(
    @Schema(description = "List of routes", required = true) List<RouteDto> routes,
    @Schema(description = "Total number of routes", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}
